staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -149,12 +165,30 @@ func (s *Validator) validateExpressions(fldPath *field.Path, sts *schema.Structu
 				// append a more descriptive error message. This error can only occur when static type checking has
 				// been bypassed. int-or-string is typed as dynamic and so bypasses compiler type checking.
 				errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("'%v': call arguments did not match a supported operator, function or macro signature for rule: %v", err, ruleErrorString(rule))))
+			} else if strings.HasPrefix(err.Error(), "operation cancelled: actual cost limit exceeded") {
+				errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("'%v': call cost exceeds limit for rule: %v", err, ruleErrorString(rule))))
 			} else {
 				// no such key: {key}, index out of bounds: {index}, integer overflow, division by zero, ...
 				errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("%v evaluating rule: %v", err, ruleErrorString(rule))))
 			}
 			continue
```

Comment:
in the error path, we `continue` and don't deduct the cost of the evaluation that errored from `remainingBudget`. that's especially surprising in the case where the evaluation errored because we exceeded our call cost

Reply:
This error occurs when the current expression running into error. I thought we should treat it similar as validation failure and keep further validation rules running as long as the overall cost budget does not exceed

Reply:
I have moved cost handle up so it would be covered. Thank you

Reply:
evalDetails should always non-nil unless CEL was configured wrong during compilation(which shouldn't happen).